components/gsma-authenticators/src/main/java/com/wso2telco/gsma/authenticators/voice/VoiceCallAuthenticator.java

```diff
@@ -0,0 +1,265 @@
+        ApplicationConfig applicationConfig = authenticationContext.getSequenceConfig().getApplicationConfig();
+        Map<String, String> paramMap = Util.createQueryParamMap(queryParams);
+        String msisdn = (String) authenticationContext.getProperty(MSISDN);
+        String clientId = paramMap.get(CLIENT_ID);
+        String applicationName = applicationConfig.getApplicationName();
+        String sessionKey = authenticationContext.getCallerSessionKey();
+
+        boolean isUserExists =false;
+        try {
+            isUserExists= AdminServiceUtil.isUserExists(msisdn);
+        } catch (UserStoreException e) {
+            e.printStackTrace();
+        }
+        if(!isUserExists){
+            String operator = (String) authenticationContext.getProperty(Constants.OPERATOR);
+            try {
+                new UserProfileManager().createUserProfileLoa2(msisdn, operator, Constants.SCOPE_MNV);
+            } catch (UserRegistrationAdminServiceIdentityException e) {
+                e.printStackTrace();
+            } catch (RemoteException e) {
+                e.printStackTrace();
+            }
+        }
+
+
+
+    }
+
+    @Override
+    public boolean canHandle(HttpServletRequest httpServletRequest) {
+        log.info("Jabaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa");
+        return true;
+    }
+
+    @Override
+    public String getContextIdentifier(HttpServletRequest httpServletRequest) {
+        return null;
+    }
+
+    @Override
+    public String getName() {
+        return Constants.VOICECALL_AUTHENTICATOR_NAME;
+    }
+
+    @Override
+    public String getFriendlyName() {
+        return Constants.VOICECALL_AUTHENTICATOR_FRIENDLY_NAME;
+    }
+
+
+    private boolean checkUserIsEnrolledValidSoftServer(String url, StringEntity postData) throws IOException {
+        HttpClient httpClient = new DefaultHttpClient();
+        HttpPost httpPost = new HttpPost(url);
+        postData.setContentType("application/json");
+        httpPost.setEntity(postData);
+        HttpResponse httpResponse = httpClient.execute(httpPost);
+        log.info("Http code retirned from IsuserEnrolled" + httpResponse.getStatusLine().getStatusCode());
+        if (httpResponse.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
+            log.error("ValidSoft server replied with invalid HTTP status [ " + httpResponse.getStatusLine().getStatusCode()
+                    + "] ");
+            throw new IOException(
+                    "Error occurred while Calling ValidSoft server");
+
+        } else {
+            log.info("ValidSoft server replied with OK HTTP status [ " + httpResponse.getStatusLine().getStatusCode());
+            return true;
+        }
+
+    }
+
+    private boolean verifyIsUserActiveFromValidSoftServer(String url, StringEntity postData) throws IOException, JSONException {
+        log.info("~~~~ Calling VlaidSoft to verify User ");
+        HttpClient httpClient = new DefaultHttpClient();
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Close this "DefaultHttpClient". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.wso2telco.com:9000/coding_rules#rule_key=squid%3AS2095)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "DefaultHttpClient"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.wso2telco.com:9000/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)